--- /usr/share/xdmod/classes/CCR/DB.php	2016-05-24 17:43:56.000000000 +0000
+++ /usr/share/xdmod/classes/CCR/DB_new.php	2017-03-29 19:11:24.393872396 +0000
@@ -65,6 +65,11 @@
     $user =     xd_utilities\getConfiguration($section, 'user');
     $passwd =   xd_utilities\getConfiguration($section, 'pass');
     $port =     xd_utilities\getConfiguration($section, 'port');
+
+    $secret = static::getHostUserPassword($host,$user);
+    if (! is_null($secret)) {
+      $passwd = $secret;
+    }
     
     $engine = "CCR\\DB\\$engine";
     
@@ -90,4 +95,27 @@
     
   }  // factory()
 
+  /**
+   * Retrieve a password for a given host, user pair
+   *
+   * @param $host  Host name as it appears in a "host" option
+   * @param $user  Username as it appears in a "user" option
+   *
+   * @throws Exception if the configuration file could not be parsed
+   *
+   * @return string  The secret, or NULL if there is no such secret
+   */
+  public static function getHostUserPassword($host,$user)
+  {
+    try {
+      $secret = xd_utilities\getConfiguration('secrets', "$user@$host");
+      return $secret;
+    } catch (Exception $e) {
+      if (strpos($e->getMessage(),'does not exist in section') === FALSE) {
+        throw $e;
+      }
+      return NULL;
+    }
+  }
+
 }  // class DB
