--- a/usr/share/xdmod/classes/OpenXdmod/Shredder/Pbs.php    2017-07-25 21:20:12.716151490 +0000
+++ b/usr/share/xdmod/classes/OpenXdmod/Shredder/Pbs.php    2017-07-25 21:51:18.891433757 +0000
@@ -156,6 +156,8 @@
         'cpus'            => 'resources_used_cpus',
     );
 
+    protected static $hostsStringToParsedHostArray;
+
     /**
      * @inheritdoc
      */
@@ -320,6 +322,11 @@
      */
     protected function parseExecHost($hosts)
     {
+        if (isset(self::$hostsStringToParsedHostArray) &&
+            array_key_exists($hosts,self::$hostsStringToParsedHostArray)) {
+           return self::$hostsStringToParsedHostArray[$hosts];
+        }
+
         $hostList = $this->parseHosts($hosts);
 
         // Key is the node name, value is the number of cpus.
@@ -343,11 +350,13 @@
             $cpuCount += $cpus;
         }
 
-        return array(
+        $parsedHostArray = array(
             'host_list'  => $hostList,
             'node_count' => $nodeCount,
             'cpu_count'  => $cpuCount,
         );
+        self::$hostsStringToParsedHostArray[$hosts] = $parsedHostArray;
+        return $parsedHostArray;
     }
 
     /**
